File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/geofence/GeofenceIsEnabledCommand.kt
package com.emarsys.emarsys_sdk.command.geofence

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class GeofenceIsEnabledCommand : EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        resultCallback.invoke(Emarsys.geofence.isEnabled(), null)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}

File: ios/Classes/Wrapper/Commands/Geofence/GeofenceisEnabledCommand.swift
//
//  Created by Emarsys on 2021. 08. 25..
//

import EmarsysSDK

class GeofenceisEnabledCommand: EmarsysCommandProtocol {
    
    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        resultCallback(["success":Emarsys.geofence.isEnabled()])
    }
    
}
